### DISCUSSION TITLE
Jpeg JXL has two modes (VarDCT and Modular), but IM seems to only use (and allow) one

### DISCUSSION THREAD

**Discussion by jumper444** [ImageMagick/ImageMagick#5437] [Development]
Jpeg JXL has two modes of operation(compression): VarDCT and Modular.
However when using IM I cannot find any setting or switch (or -define method) that would allow selecting one or the other.

1) Is there a method to specify which compression to use (in a 'convert' for example)?
2) If not, then which of the two is IM defaulting to?
3) And of course would both be added in the future?  They are in the reference lib as far as I know and already available in common image programs like Irfanview.  The two compression types differ significantly depending on quality and image type so this would be useful (Modular mode is much better for compressing line art, bitmap screenshot, etc...stuff without gradients and including document scans.)

**Comment by joppuyo** [comment]
[Looking at the source code](https://github.com/ImageMagick/ImageMagick/blob/1c7af5461d4defbcfcac2a8db3e4fb1a81324861/coders/jxl.c#L873-L875 ), it seems like specifying quality of 100 will produce a lossless and therefore modular file.

jxlinfo doesn't currently expose information if the file is modular or vardct but I tested compressing a simple graphic both at Q100 and Q99 and the Q99 version has a significantly larger file size so I'm going to assume the Q100 one is a modular image.

A quality of 99 or less seems to produce a vardct image.

So currently it seems like we are limited either to a lossless modular file or a lossy vardct file, lossy modular doesn't seem to be supported at the moment.
